torch/distributed/checkpoint/state_dict.py

```diff
@@ -747,6 +747,11 @@ def _device(t):
             flatten_osd, osd_mapping = _flatten_state_dict(optim_state_dict)
             flatten_local_osd, local_osd_mapping = _flatten_state_dict(local_state_dict)
             _broadcast_state_dict(flatten_osd, flatten_local_osd, device=device)
+            for optim_key in flatten_osd.keys():
+                if optim_key not in flatten_local_osd:
```

Comment:
I'm wondering when and why this happen. Why `initial_lr` does not exist in the loca_osd.